src/mscorlib/src/System/String.Comparison.cs

```diff
@@ -1126,7 +1018,10 @@ public Boolean StartsWith(String value, StringComparison comparisonType)
                     }
                     return (value.Length == 1) ?
                             true :                 // First char is the same and thats all there is to compare
-                            StartsWithOrdinalHelper(this, value);
+                            SpanHelpers.SequenceEqual(
```

Comment:
What happens if this.Length is less than value.Length?

Reply:
This condition is checked two lines above this one.